src/launcher/Launcher.cc

```diff
@@ -244,3 +245,28 @@ void Launcher::startEditor(Fl_Widget* btn, void* userdata) {
 	window->writeJsonFile();
 	window->startExecutable(true);
 }
+
+void Launcher::guessVersion(Fl_Widget* btn, void* userdata) {
+	Launcher* window = static_cast< Launcher* >( userdata );
+    auto choice = fl_choice("Comparing resources packs can take a long time.\nAre you sure you want to continue?", "Stop", "Continue", 0);
+    if (choice != 1)
+        return;
+
+    char* log = NULL;
+    if (guess_resource_version(window->engine_options, &log)) {
```

Comment:
Add `window->writeJsonFile();` here in order for detection to use the currently selected directory.

Reply:
I see, the problem is that engine_options is only updated when you launch the game or the editor.

Since changing the input fields doesn't update the json, then guessing the version shouldn't update it either. I'll change the rust-C api instead.